samples/web_shell/lib/shell/service_worker.js

```diff
@@ -36,8 +38,13 @@ export default ServiceWorker;
  */
 ServiceWorker.prototype.register = function() {
   if ('serviceWorker' in navigator) {
-    return navigator.serviceWorker.register('/service_worker.js',
-        {scope: '/'}).then(function(reg) {
+
+    var curloc = document.location.href.split('/');
+    curloc = curloc.splice(0, curloc.length - 1).join('/');
```

Comment:
As discussed, `slice` is easier to understand, and `splice` seems behaving suspiciously in Chrome.